lib/honeycomb/propagation/w3c.rb

```diff
@@ -42,14 +42,36 @@ def parse_v1(payload)
 
     # Serialize trace headers
     module MarshalTraceContext
-      def to_trace_header
+      TRACE_ID_REGEX = /^[A-Fa-f0-9]{32}$/.freeze
+      SPAN_ID_REGEX = /^[A-Fa-f0-9]{16}$/.freeze
+
+      def to_trace_header(propagation_context: nil)
+        trace_id, span_id = propagation_context
         # do not propagate malformed ids
-        if trace.id =~ /^[A-Fa-f0-9]{32}$/ && id =~ /^[A-Fa-f0-9]{16}$/
-          return "00-#{trace.id}-#{id}-01"
+        if trace_id =~ TRACE_ID_REGEX && span_id =~ SPAN_ID_REGEX
```

Comment:
this may be breaking since this is already exported. I think I'll add a fallback as a precaution

Reply:
added!